### PR TITLE
feat: add `update` method as safer alternative to `overwrite`

### PR DESCRIPTION
**Pull request by benmccann**
In https://github.com/vitejs/vite/pull/7397 we added `{ contentOnly: true }` to every call to `overwrite` in order to fix https://github.com/vitejs/vite/issues/7365. This could probably be the default. To quote @dummdidumm on that issue:

>The `overwrite` call removes all prepends/appends that were done on the start/end position if you don't pass in the `{ contentOnly: true }` option as forth argument. I never came across a case where I would want the default behavior, so I'm always setting that option.

### CODE REVIEW

**Comment by antfu** [comment]
Instead of deprecating it, I feel there is no harm to provide both options for different usage. We could just update the examples to prompt `.update` more.

**Comment by benmccann** [comment]
Yes, good idea. I've made that change. Sorry for the delay - this one got lost in my inbox

**Comment by antfu** [comment]
Can you also update the dts? Thanks

**Comment by antfu** [comment]
Also would be great to include a few tests for it.

File: README.md [hunk 1]
```diff
@@ -149,6 +153,8 @@ Moves the characters from `start` and `end` to `index`. Returns `this`.
 
 ### s.overwrite( start, end, content[, options] )
 
+__Note:__ It may be preferred to use `s.update(...)` instead
```
Comment (antfu) [inline README.md:156]: Can we have a brief introduction of why?

Comment (benmccann) [reply README.md]: done

**Review by antfu — APPROVED** [review]

File: src/MagicString.js [hunk 2]
```diff
@@ -349,6 +349,12 @@ export default class MagicString {
 	}
 
 	overwrite(start, end, content, options) {
+		options = options || {};
+		options.overwrite = !options.contentOnly;
```
Comment (jdalton) [inline src/MagicString.js:353]: Modifying a user provided options object isn't great DX and unexpected.

Comment (antfu) [reply src/MagicString.js:353]: Sure, PR welcome

Comment (benmccann) [reply src/MagicString.js:353]: yes, you're right. I've sent a PR: https://github.com/Rich-Harris/magic-string/pull/227

**Pull request merged**